### PR TITLE
SIRo Fixes for Multi-agent

### PR DESCRIPTION
**Pull request by ASzot**
Changes to get multi-agent training working. See the added `multi_agent_kinematic_oracle.yaml` to run multi-agent training.

### CODE REVIEW

File: habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py [hunk 1]
```diff
@@ -175,17 +178,21 @@ def _sample_active(self):
...
         )
 
-        self._multi_storage.set_active(
-            [self._agents[i].rollouts for i in self._active_agents]
-        )
-        self._multi_updater.set_active(
-            [self._agents[i].updater for i in self._active_agents]
-        )
+        if not self._is_post_init:
+            # If not post init then we are running in evaluation mode and
```
Comment (vincentpierre) [inline habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py:182]: It seems that the code does `if NOT _is_post_init then initialize storage`, but the comment you added says that `if NOT _is_post_init then DON'T initialize`. I may be missing something, could you clarify ?


This feels strange. You are assuming how the code will be called. I would log a warning here.
"_get_agents was called before post_init. As such, only the policy will be setup and not the rollout storage. This is expected in evaluation mode"

File: habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py [hunk 2]
```diff
@@ -273,4 +282,31 @@ def updater(self):
...
+                        policy_i * hxs_dim, (policy_i + 1) * hxs_dim
+                    )
+                    rnn_hxs[env_i, :, rnn_sel] = action_data.rnn_hidden_states[
+                        env_i, :, rnn_sel
+                    ]
+
+                    ac_sel = slice(policy_i * ac_dim, (policy_i + 1) * ac_dim)
+                    prev_actions[env_i, ac_sel].copy_(
+                        action_data.actions[env_i, ac_sel]
+                    )
```
Comment (vincentpierre) [inline habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py:312]:
```suggestion
                if not agent_should_insert.item():
                    continue
                rnn_sel = slice(
                    policy_i * hxs_dim, (policy_i + 1) * hxs_dim
                )
                rnn_hxs[env_i, :, rnn_sel] = action_data.rnn_hidden_states[
                    env_i, :, rnn_sel
                ]

                ac_sel = slice(policy_i * ac_dim, (policy_i + 1) * ac_dim)
                prev_actions[env_i, ac_sel].copy_(
                    action_data.actions[env_i, ac_sel]
                )
```
The code is already very nested. Consider using continue (similar to the code above) to reduce nesting.

**Review by vincentpierre — APPROVED** [review]

**Review by akshararai — APPROVED** [review]
some clarification about `self._is_post_init` but looks good otherwise.

File: habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py (hunk 1 above)
Comment (ASzot) [reply habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py:215]: Oops, yeah this was a mistake.

**Pull request merged**